src/jackdaw/test/fixtures.clj

```diff
@@ -55,6 +57,42 @@
        (log/info "topic-fixture: created topics: " (keys topic-config))
        (t)))))
 
+;;; application reset ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn delete-files-recursively [fname & [silently]]
+  (letfn [(delete-f [file]
+            (when (.isDirectory file)
+              (doseq [child-file (.listFiles file)]
+                (delete-f child-file)))
+            (clojure.java.io/delete-file file silently))]
+    (delete-f (clojure.java.io/file fname))))
+
+(defn reset-application-fixture [app-config]
+  (fn [t]
+    (let [rt (StreamsResetter.)
+          app-id (get app-config "application.id")
+          app-state (format "%s/%s"
+                            (or (get app-config "state.dir")
+                                "/tmp/kafka-streams")
+                            app-id)]
+      (when (.exists (io/as-file app-state))
+        (delete-files-recursively app-state))
```

Comment:
The app should not be running at the point of this function being called. But @gphilipp thought this might be better as an option to the kstream-fixture that could just use the `.cleanUp` method of the kstream it has just created (but not yet started). A similar option on the `topic-fixture` could be used to trigger topic-deletion before re-creation.